Unable to compile against OpenSSL 4.0

**Describe the bug**
Unable to compile rauc 1.15.2 against OpenSSL 4.0

`error: invalid use of incomplete typedef ‘ASN1_OCTET_STRING’ {aka ‘struct asn1_string_st’}`

**Background information**
Build rauc 1.15.2 on Fedora 45 (rawhide).


**To Reproduce**
RPM spec:
https://src.fedoraproject.org/rpms/rauc/blob/rawhide/f/rauc.spec

**Expected behavior**
Possible to build rauc against OpenSSL 4.0.

**Logs**
Multiple errors and warnings when openssl was upgraded from 3.5.5 to 4.0.0.

Build log:
https://kojipkgs.fedoraproject.org/work/tasks/3095/144223095/build.log

Root log:
https://kojipkgs.fedoraproject.org/work/tasks/3095/144223095/root.log

```
../src/signature.c: In function ‘load_key_pkcs11’:
../src/signature.c:268:1: warning: label ‘out’ defined but not used [-Wunused-label]
  268 | out:
      | ^~~
../src/signature.c: In function ‘load_cert_pkcs11’:
../src/signature.c:408:1: warning: label ‘out’ defined but not used [-Wunused-label]
  408 | out:
      | ^~~
../src/signature.c: In function ‘cms_sign’:
../src/signature.c:681:25: warning: ‘X509_cmp_current_time’ is deprecated: Since OpenSSL 4.0 [-Wdeprecated-declarations]
  681 |                         if (X509_cmp_current_time(expiry_time) == 1 && X509_cmp_time(expiry_time, &comp) == -1) {
      |                         ^~
In file included from /usr/include/openssl/cms.h:29,
                 from ../src/signature.c:2:
/usr/include/openssl/x509.h:694:27: note: declared here
  694 | OSSL_DEPRECATEDIN_4_0 int X509_cmp_current_time(const ASN1_TIME *s);
      |                           ^~~~~~~~~~~~~~~~~~~~~
../src/signature.c:681:25: warning: ‘X509_cmp_time’ is deprecated: Since OpenSSL 4.0 [-Wdeprecated-declarations]
  681 |                         if (X509_cmp_current_time(expiry_time) == 1 && X509_cmp_time(expiry_time, &comp) == -1) {
      |                         ^~
/usr/include/openssl/x509.h:693:27: note: declared here
  693 | OSSL_DEPRECATEDIN_4_0 int X509_cmp_time(const ASN1_TIME *s, const time_t *t);
      |                           ^~~~~~~~~~~~~
../src/signature.c: In function ‘cms_check_signer_cns’:
../src/signature.c:1172:45: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1172 |                 X509_NAME *current_signer = X509_get_subject_name(sk_X509_value(signers, i));
      |                                             ^~~~~~~~~~~~~~~~~~~~~
../src/signature.c: In function ‘cms_get_unverified_manifest’:
../src/signature.c:1403:24: error: invalid use of incomplete typedef ‘ASN1_OCTET_STRING’ {aka ‘struct asn1_string_st’}
 1403 |         if (!(*content)->data || ((*content)->length <= 0)) {
      |                        ^~
../src/signature.c:1403:45: error: invalid use of incomplete typedef ‘ASN1_OCTET_STRING’ {aka ‘struct asn1_string_st’}
 1403 |         if (!(*content)->data || ((*content)->length <= 0)) {
      |                                             ^~
../src/signature.c:1412:37: error: invalid use of incomplete typedef ‘ASN1_OCTET_STRING’ {aka ‘struct asn1_string_st’}
 1412 |         tmp = g_bytes_new((*content)->data, (*content)->length);
      |                                     ^~
../src/signature.c:1412:55: error: invalid use of incomplete typedef ‘ASN1_OCTET_STRING’ {aka ‘struct asn1_string_st’}
 1412 |         tmp = g_bytes_new((*content)->data, (*content)->length);
      |                                                       ^~
../src/signature.c: In function ‘cms_verify_bytes’:
../src/signature.c:1518:20: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
 1518 |                 so = X509_ATTRIBUTE_get0_type(xa, 0);
      |                    ^
```

--- a/src/signature.c
+++ b/src/signature.c
@@ -1344,7 +1344,7 @@ gboolean cms_get_unverified_manifest(GBytes *sig, GBytes **manifest, GError **er
 				"missing manifest in inline signature");
 		goto out;
 	}
-	if (!(*content)->data || ((*content)->length <= 0)) {
+	if ((ASN1_STRING_get0_data(*content) == NULL) || (ASN1_STRING_length(*content) <= 0)) {
 		g_set_error(
 				error,
 				R_SIGNATURE_ERROR,
@@ -1353,7 +1353,7 @@ gboolean cms_get_unverified_manifest(GBytes *sig, GBytes **manifest, GError **er
 		goto out;
 	}
 
-	tmp = g_bytes_new((*content)->data, (*content)->length);
+	tmp = g_bytes_new(ASN1_STRING_get0_data(*content), ASN1_STRING_length(*content));
 	if (!tmp) {
 		g_set_error_literal(
 				error,

